app/cdap/components/GridTable/index.tsx

```diff
@@ -233,10 +256,48 @@ export default function GridTable() {
     setRowsDataList(rowData);
   };
 
+  const isParsingPanel =
+    dataprep?.insights?.name &&
+    isFirstWrangle &&
+    connectorType === 'File' &&
+    Array.isArray(gridData?.headers) &&
+    gridData?.headers.length !== 0;
+
   useEffect(() => {
     getGridTableData();
   }, [gridData]);
 
+  const addDirectives = (directive: string) => {
+    setLoading(true);
+    if (directive) {
+      const apiPayload: IApiPayload = getAPIRequestPayload(params, directive, '');
+      addDirectiveAPICall(apiPayload);
+    }
+  };
+
+  const addDirectiveAPICall = (apiPayload: IApiPayload) => {
```

Comment:
as the file size is greater than 200 lines, can we move addDirectiveAPICall to a separate file called utils.ts
and just return the final states in an object and update the states here inside the component